### ISSUE TITLE
Azure Portal: Remove "Enable Preview" from channel settings

### ISSUE THREAD

**Issue by compulim** [Microsoft/BotFramework-WebChat#1086]
- [ ] Remove "Enable preview" check box
- [ ] Add a new "Use deprecated version" check box
- [ ] By default, will load the latest version (v4)

**Comment by compulim** [comment]
@seaen

- Can you rephrase "use deprecated version"?
- Can you write us a story on this one? E.g. Adaptive Cards compatibility

**Comment by compulim** [comment]
This is my main feature for 4.4. There are lots of discussions between Web Chat and the service team about this issue. The design has been flushed out and ready to implement.

**Comment by corinagum** [comment]
Adding to discussion from #1885 by @mahomedalid

> The   Azure Bot Registration section displays an iframe embed code as a suggested way to solve an integrated webchat on your site, but seems ConversationUpdate is not being triggered because it doesn't inject the user id by default. 
> I see there is a similar issue (#752) but the fact that the iframe is the "recommended way" (at least from the azure portal perspective) make me think that we can improve the experience, maybe suggesting a snippet of code using webchat.


@compulim, here are some additional questions I want to add on.
1. Can you give a general summary of how the service side's Web Chat will be changing? How will users, for example, pass in user id to trigger ConversationUpdate?
1. Will `iframe` still be used? What about secrets vs tokens, and versioning?
1. Do you have other details about pending changes that customers would be interested in?

I think once the major update is made, we should also discuss how and where we will keep documentation related to ABS side of Web Chat. Right now we don't have much documentation about ABS Web Chat because it is v3, especially on this repo, but we should work to improve that.

**Comment by compulim** [comment]
1. `conversationUpdate` will still the same as of today
   - There will be no way to send custom welcome message, anyone interested, please open a bug to discuss
   - (New) Like today, developers can specify `userid`, if it is conflict with JWT token, we will use the one from JWT token
   - Like today, if `userid` is not specified, we will randomly create one and store it in cookie of webchat.botframework.com
2. Yes, we are still `iframe`. Our target scenario is still the same, "static web site". After this work is done, we will talk to channel team about using HTTP referrer to secure secret
   - Versioning, we support "latest", "fixed major version", and "fixed minor version". We do not support "fixed patch version"
      - It is very likely we will support fixed version for 2 years after they go production
      - After 2 years, we will bump automatically
      - Developers can keep using the "old v3" versions, by specifying `ibiza` as the version number
   - We will roll out new versions gradually (A/B on percentage)
3. Few interesting things
   - This rework will enable us to provide more functionalities easily, please open bugs for feature suggestions
      - `postMessage` API, e.g. controlling greedy focus
      - Setting `styleOptions` via JSON file
      - Minimizable UI with proactive messaging support
   - We will enable OAuth, but magic code is always required due to security reason
   - It is very likely we will put the source code of the embed page on GitHub
   - It is very likely for a new REST API, used by the embed page, to obtain bot configurations, including bot name, bot icon URL, etc
   - Bot avatar will be shown in v4 automatically if bot icon is uploaded in Azure Portal
   - Using v4, we will not show a title bar (same as CDN/NPM), developers are advised to wrap the `<iframe>` with a `<div>` to show the title, this is because the customization of title varies a lot
   - Greedy focus is removed in v4, but still be provided in v3

**Comment by compulim** [comment]
This is done. We will switching it on gradually over the next few weeks.
